### DISCUSSION TITLE
SVG Rendering questions

### DISCUSSION THREAD

**Discussion by baleksey** [rust-skia/rust-skia#928] [Q&A]
Skia SVG render is good but I've stumbled upon several roadblocks and I would be appreciated for help.

Here is simplest code to draw SVG on the Canvas:
``` 
    use std::{fs::File, io::Write, path::Path};
    use super::Dom;
    use crate::{modules::svg::decode_base64, surfaces, EncodedImageFormat, FontMgr, Surface};
    fn render_simple_svg() {

        let svg = r##"<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100" height = "256" width = "256">
            <path d="M30,1h40l29,29v40l-29,29h-40l-29-29v-40z" stroke="#;000" fill="none"/>
            <path d="M31,3h38l28,28v38l-28,28h-38l-28-28v-38z" fill="#a23"/>
            <text x="50" y="68" font-size="48" fill="#FFF" text-anchor="middle"><![CDATA[410]]></text>
            </svg>"##;

        let mut surface = surfaces::raster_n32_premul((256, 256)).unwrap();
        let canvas = surface.canvas();
        let font_mgr = FontMgr::new();

        let dom = Dom::from_str(svg, font_mgr).unwrap();

        dom.render(canvas);
    }
```

**Questions:**
1. What is the easiest built-in way to get width/height/viewbox from loaded (external) dom element? 
2. How can I render SVG on canvas with set_alpha (so svg appears transparent)? Current lifehack is to add opacity="0.5" to root svg element, but it requires parsing from string every time (and I need that a lot). Is there better way to set params in the dom or add "paint" to render?
3. The same question with Blending Modes. How to render svg with "Overlay" (or other) blending mode? Workaround is to render svg on temp canvas then draw temp on our main, but it took 3-10x more time then just rendering SVG without any temp canvases and my software must be as fast as possible. Is there any way I can do it faster?

**Comment by pragmatrix** [comment]
Can you take a look if a Rust wrapper of `SkSVGDom::getRoot()` and [SkSVGNode](https://api.skia.org/classSkSVGNode.html) would be sufficient to resolve the issues you described. If so, I suppose, it would be a good step forward to wrap the SkSVGNode first and then add more of the node type hierarchy as needed.
